src/DurableTask.Netherite/OrchestrationService/Client.cs

```diff
@@ -105,6 +108,8 @@ public void Process(ClientEvent clientEvent)
 
                     var reassembledEvent = FragmentationAndReassembly.Reassemble<ClientEvent>(stream, fragment);
 
+                    this.Fragments.Remove(group);
```

Comment:
This appears to be new behavior. Is this part of the bug fix?

Reply:
I moved this here from line 136 (which was executed by the call on the next line). It should be logically equivalent.